MonitorControl/Model/Display.swift

```diff
@@ -62,15 +63,16 @@ class Display: Equatable {
     (key ?? PrefKey.value).rawValue + (command != nil ? String((command ?? Command.none).rawValue) : "") + self.prefsId
   }
 
-  internal init(_ identifier: CGDirectDisplayID, name: String, vendorNumber: UInt32?, modelNumber: UInt32?, isVirtual: Bool = false, isDummy: Bool = false) {
+  init(_ identifier: CGDirectDisplayID, name: String, vendorNumber: UInt32?, modelNumber: UInt32?, serialNumber: UInt32?, isVirtual: Bool = false, isDummy: Bool = false) {
     self.identifier = identifier
     self.name = name
     self.vendorNumber = vendorNumber
     self.modelNumber = modelNumber
-    self.prefsId = "(" + String(name.filter { !$0.isWhitespace }) + String(vendorNumber ?? 0) + String(modelNumber ?? 0) + "@" + String(identifier) + ")"
-    os_log("Display init with prefsIdentifier %{public}@", type: .info, self.prefsId)
+    self.serialNumber = serialNumber
     self.isVirtual = DEBUG_VIRTUAL ? true : isVirtual
     self.isDummy = isDummy
+    self.prefsId = "(" + String(name.filter { !$0.isWhitespace }) + String(vendorNumber ?? 0) + String(modelNumber ?? 0) + "@" + (self.isVirtual ? String(self.serialNumber ?? 9999) : String(identifier)) + ")"
```

Comment:
Why not use string concatenation here instead? I think that might simplify this line quite a bit.

I haven't really tested it myself, because I don't really know how the virtual display feature is supposed to work, but will this handle me changing the name of the display without issues? Or is this name property a different one from the one the user can configure?

Reply:
That looks nicer.

Yes, this name is a permanent one and different from friendlyName which was differentiated a while back.

Reply:
Fixed that line to look nice. Approve please! Thanks! :)